hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/PrehandleHandlerContext.java

```diff
@@ -29,63 +30,141 @@
 import edu.umd.cs.findbugs.annotations.Nullable;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Objects;
 
 /**
- * Base abstract class for constructing {@link TransactionMetadata} by collecting information that
- * is needed when transactions are handled as part of "pre-handle" needed for signature
- * verification.
- *
- * <p>NOTE : This class is designed to be subclassed
+ * Interface for constructing {@link TransactionMetadata} by collecting information that is needed
+ * when transactions are handled as part of "pre-handle" needed for signature verification.
  */
-public abstract class TransactionMetadataBuilder<T extends TransactionMetadataBuilder<T>> {
-    protected final List<HederaKey> requiredNonPayerKeys = new ArrayList<>();
-    protected HederaKey payerKey;
-    protected ResponseCodeEnum status = OK;
-    protected TransactionBody txn;
-    protected final AccountKeyLookup keyLookup;
-    protected AccountID payer;
-
-    protected TransactionMetadataBuilder(@NonNull final AccountKeyLookup keyLookup) {
-        this.keyLookup = Objects.requireNonNull(keyLookup);
+public class PrehandleHandlerContext {
+
+    private final AccountKeyLookup keyLookup;
+
+    private final TransactionBody txn;
+    private final AccountID payer;
+    private final List<HederaKey> requiredNonPayerKeys = new ArrayList<>();
+
+    private ResponseCodeEnum status = OK;
+    private HederaKey payerKey;
+    private Object handlerMetadata;
```

Comment:
I think `TransactionMetadata` record is clearer than Object

Reply:
Currently we don't have any other transactions other than schedules. Object is not clear.

Reply:
But we have discussed that in one of our meetings. We agreed to use `Object` or a marker interface to be flexible.